src/public.js

```diff
@@ -39,7 +39,9 @@ const hasToken = (document.getElementById('sharingToken') !== null);
 const isPasswordProtected = (document.getElementById('password-submit') !== null);
 
 if (hasToken && !isPasswordProtected) {
-	$('body').prepend($('<div>').attr('id', 'terms_of_service_confirm'));
+	const tofc = document.createElement('div');
+	tofc.id = 'terms_of_service_confirm';
```

Comment:
In other apps we use `.setAttribute('id', 'notifications')`
https://github.com/nextcloud/notifications/blob/master/src/Init.js#L31

But seems like they both work?